macaroons/store.go

```diff
@@ -78,7 +77,10 @@ func NewRootKeyStorage(db kvdb.Backend) (*RootKeyStorage, error) {
 	}
 
 	// Return the DB wrapped in a RootKeyStorage object.
-	return &RootKeyStorage{Backend: db, encKey: nil}, nil
+	return &RootKeyStorage{
+		Backend: db,
```

Comment:
What I ran into is that this line 

https://github.com/lightningnetwork/lnd/blob/fb17bf2bb5cd49c01501e1842ed8082df75d5877/macaroons/store.go#L100

should not pass in `r`, but `r.Backend`. Otherwise the extended backend isn't recognized. Best to panic in `BeginReadTx` to always catch these things.

Reply:
Ah yes, makes sense. Good catch!

You mean to add a panic to `BeginReadTx` of the etcd and postgres implementations because there we always expect an extended DB? Or where exactly would you put that panic?

Reply:
Yes, I put it there indeed.

Reply:
Also with the dropped routing tx commit, there is still a BeginReadTx in `newDbRoutingTx`

Reply:
I went ahead and implemented the idea suggested by @bhandras to add the `View()` and `Update()` methods directly to the `walletdb.DB` interface. That way we don't do any runtime interface checks anymore which removes the need for a panic.

A manual `BeginReadTx` is okay I think since that cannot cause another TX to fail as it doesn't change any data.

Reply:
I remember reading about serializable isolation and iirc there was also something with a read failing under certain conditions.

Reply:
@guggero are you sure that read-only txes cannot fail? I'd think that a db engine can overlap txes in such a way that a read-only has to give way to a write.

Reply:
I believe this is where I saw it: https://wiki.postgresql.org/wiki/SSI

Last example, "rollover"

Reply:
Hmm, good point. We are going to address the issue of manual `BeginReadTx` in https://github.com/lightningnetwork/lnd/pull/5595 though.